pkg/datatransfers/data_transfers.go

```diff
@@ -81,243 +77,177 @@ MoveKeys performs physical key-range move from one datashard to another.
 It is assumed that passed key range is already locked on every online spqr-router.
 
 Steps:
-  - traverse pg_class to resolve all relations that matches given sharding rules
-  - create sql copy and delete queries to move data tuples.
-  - prepare and commit distributed move transaction
+  - create postgres_fdw on receiving shard
+  - copy data from sending shard to receiving shard via fdw
+  - delete data from sending shard
 */
-func MoveKeys(ctx context.Context, fromId, toId string, keyr qdb.KeyRange, db qdb.XQDB) error {
+func MoveKeys(ctx context.Context, fromId, toId string, krg *kr.KeyRange, ds *distributions.Distribution, db qdb.XQDB, cr coordinator.Coordinator) error {
+	tx, err := db.GetTransferTx(ctx, krg.ID)
+	if err != nil {
+		return err
+	}
+	if tx == nil {
+		tx = &qdb.DataTransferTransaction{
+			ToShardId:   toId,
+			FromShardId: fromId,
+			Status:      qdb.Planned,
+		}
+		if err = db.RecordTransferTx(ctx, krg.ID, tx); err != nil {
+			return err
+		}
+	}
 	if shards == nil {
 		err := LoadConfig(config.CoordinatorConfig().ShardDataCfg)
 		if err != nil {
 			spqrlog.Zero.Error().Err(err).Msg("error loading config")
 		}
 	}
 
-	from, err := pgx.Connect(ctx, createConnString(fromId))
+	from, err := pgx.Connect(ctx, createConnString(fromId)) // nolint: staticcheck
```

Comment:
Why not to fix it?

Reply:
It's a false positive

Reply:
Lint magically stoped failing here, so I removed the nolint